code/datums/components/pet_collar.dm

```diff
@@ -0,0 +1,208 @@
+	if(!collar)
+		return
+
+	collar.forceMove(new_loc)
+	collar = null
+
+	if(collar_icon_state && update_visuals)
+		update_overlays()
+
+/**
+ * Update the collar overlay.
+ */
+/datum/component/pet_collar/proc/update_overlays()
+	var/mob/living/parent_mob = parent
+
+	if(applied_collar_overlays)
+		parent_mob.cut_overlay(applied_collar_overlays)
+		applied_collar_overlays = null
+
+	if(collar && collar_icon_state)
+		var/stat_tag = parent_mob.stat == DEAD ? "_dead" : (parent_mob.resting ? "_rest" : "")
+
+		applied_collar_overlays = list(
+			"[collar_icon_state][stat_tag]collar",
+			"[collar_icon_state][stat_tag]tag",
+		)
+
+		parent_mob.add_overlay(applied_collar_overlays)
+
+/**
+ * Handler for COMSIG_ATOM_CONTENTS_DEL
+ */
+/datum/component/pet_collar/proc/on_handle_atom_del(mob/parent_mob, atom/deleting_atom)
+	SIGNAL_HANDLER
+
+	if(deleting_atom != collar)
+		return
+
+	collar = null
+
+	if(QDELETED(parent_mob))
+		return
+
+	update_overlays()
+
+/**
+ * Handler for COMSIG_LIVING_GIBBED
+ */
+/datum/component/pet_collar/proc/on_gib(mob/parent_mob, no_brain, no_organs, no_bodyparts)
+	SIGNAL_HANDLER
+
+	remove_collar(parent_mob.drop_location(), update_visuals = FALSE)
+
+/**
+ * Handler for COMSIG_LIVING_REVIVE and COMSIG_LIVING_DEATH
+ */
+/datum/component/pet_collar/proc/on_life_change(mob/parent_mob)
+	SIGNAL_HANDLER
+
+	update_overlays()
+
+/**
+ * Handler for COMSIG_PARENT_ATTACKBY
+ */
+/datum/component/pet_collar/proc/on_attackby(mob/parent_mob, obj/item/thing, mob/user)
+	SIGNAL_HANDLER
+
+	if(istype(thing, /obj/item/clothing/neck/petcollar) && !collar)
+		add_collar(thing, user)
+		return COMPONENT_NO_AFTERATTACK
+
+	return NONE
+
+
+/**
+ * A corresponding strippable_item for the collar slot.
+ * Note that this shouldn't be used for carbons as it overlaps the carbon neck slot.
+ */
+/datum/strippable_item/pet_collar
+	key = STRIPPABLE_ITEM_PET_COLLAR
+
+/datum/strippable_item/pet_collar/get_item(atom/source)
+	var/datum/component/pet_collar/collar_component = source?.GetComponent(/datum/component/pet_collar)
+	if(!collar_component)
+		return
+
+	return collar_component.collar
+
+/datum/strippable_item/pet_collar/try_equip(atom/source, obj/item/equipping, mob/user)
+	. = ..()
+	if (!.)
+		return FALSE
+
+	if (!istype(equipping, /obj/item/clothing/neck/petcollar))
+		to_chat(user, span_warning("That's not a collar."))
+		return FALSE
+
+	return TRUE
+
+/datum/strippable_item/pet_collar/finish_equip(atom/source, obj/item/equipping, mob/user)
+	var/datum/component/pet_collar/collar_component = source?.GetComponent(/datum/component/pet_collar)
+	if(!collar_component)
+		return
+
+	collar_component.add_collar(equipping, user)
+
+/datum/strippable_item/pet_collar/finish_unequip(atom/source, mob/user)
+	var/datum/component/pet_collar/collar_component = source?.GetComponent(/datum/component/pet_collar)
+	if(!collar_component)
+		return
+
+	var/obj/collar = collar_component.remove_collar(user.drop_location())
```

Comment:
This feels like a really bad component

Reply:
Mmm this gives me some pause but I'm not actually sure it's that bad or if the alternatives are any better, what do you have in mind here?

Reply:
we have the updates_icon_onmob element for handling the icon stuff.

Reply:
i agree with kapu, this shouldn't be necessary

Reply:
Is the main hangup that I moved an inventory slot there? I can move it back to non-component code in pet class if you prefer.

Reply:
> I don't really see how `/datum/element/update_icon_updates_onmob` would be useful here, collars don't change their icon themselves, and it calls functions that are only relevant to carbons.

`update_icon()` can send the relevant signal to it's owner. If the owner is a carbon, it will update the specified slot (ITEM_SLOT_NECK), and if its not a carbon it will just call their `update_icons`. 

The issue here is that the componentization is bad. someone else can explain in detail why.

Reply:
I don't see how that's really needed, the pet collar itself doesn't update icons, just the owner does.

Reply:
Alright, I'll switch back to it being part of the pet class since I don't think there's any other way to make it work with the `/datum/strippable_item` without referencing it like I am.

Reply:
It's now part of both pet classes again. Retested functionality.